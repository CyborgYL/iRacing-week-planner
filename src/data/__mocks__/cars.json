[
  {
    "freeWithSubscription": false,
    "skuname": "ARCA Menards Chevrolet Impala",
    "name": "ARCA Menards Chevrolet Impala",
    "sku": 10088,
    "discountGroupNames": [
      "oval car"
    ]
  },
  {
    "freeWithSubscription": false,
    "skuname": "Aston Martin DBR9 GT1",
    "name": "Aston Martin DBR9 GT1",
    "sku": 10389,
    "discountGroupNames": [
      "road car"
    ]
  },
  {
    "freeWithSubscription": false,
    "skuname": "Audi 90 GTO",
    "name": "Audi 90 GTO",
    "sku": 10410,
    "discountGroupNames": [
      "road car"
    ]
  },
  {
    "freeWithSubscription": false,
    "skuname": "Audi R18",
    "name": "Audi R18",
    "sku": 10439,
    "discountGroupNames": [
      "road car"
    ]
  },
  {
    "freeWithSubscription": false,
    "skuname": "Audi R8 LMS",
    "name": "Audi R8 LMS",
    "sku": 10405,
    "discountGroupNames": [
      "road car"
    ]
  },
  {
    "freeWithSubscription": false,
    "skuname": "Audi RS 3 LMS",
    "name": "Audi RS 3 LMS",
    "sku": 10461,
    "discountGroupNames": [
      "road car"
    ]
  },
  {
    "freeWithSubscription": false,
    "skuname": "BMW M4 GT3",
    "name": "BMW M4 GT3",
    "sku": 10494,
    "discountGroupNames": [
      "road car"
    ]
  },
  {
    "freeWithSubscription": false,
    "skuname": "BMW M4 GT4",
    "name": "BMW M4 GT4",
    "sku": 10479,
    "discountGroupNames": [
      "road car"
    ]
  }
]
