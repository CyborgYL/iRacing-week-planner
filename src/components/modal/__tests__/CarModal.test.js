
import { describe, test } from '@jest/globals';
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import CarModal from '../CarModal';

import '../../../data/cars.json';

jest.mock('../../../data/cars.json');

describe('components/modal/CarModal', () => {
  beforeAll(() => {
    ReactDOM.createPortal = jest.fn((element, node) => {
      return element;
    });
  })

  afterEach(() => {
    ReactDOM.createPortal.mockClear();
  })

  test('renders correctly', () => {
    const onClose = jest.fn();
    const component = renderer.create(
      <CarModal
        onClose={onClose}
        ownedCars={[10494, 10410]}
        favouriteCars={[10461, 10410]}
        isOpen={true}
        carIds={[10410, 10494, 10461, 10405]}
        seriesName="Audi Spectacular"
      />,
    );

    expect(component.toJSON()).toMatchSnapshot();

    const closeButton = component.root.findByProps({ className: 'close' });
    closeButton.props.onClick({ preventDefault: () => {}, stopPropagation: () => {}});
    expect(onClose).toBeCalled();


  });
});
