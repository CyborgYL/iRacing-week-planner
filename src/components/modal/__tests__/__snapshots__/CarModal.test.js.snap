// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/modal/CarModal renders correctly 1`] = `
<div
  className="ReactModal__Overlay"
  onClick={[Function]}
  onMouseDown={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(255, 255, 255, 0.75)",
      "bottom": 0,
      "left": 0,
      "position": "fixed",
      "right": 0,
      "top": 0,
    }
  }
>
  <div
    aria-modal={true}
    className="ReactModal__Content Modal__Bootstrap modal-dialog modal-lg"
    onClick={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onMouseUp={[Function]}
    role="dialog"
    style={Object {}}
    tabIndex="-1"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <button
          aria-label="Close"
          className="close"
          onClick={[Function]}
          type="button"
        >
          <span
            className="glyphicon glyphicon-remove"
          />
        </button>
        <h4
          className="modal-title"
        >
          Cars for Audi Spectacular
        </h4>
      </div>
      <div
        className="modal-body"
        style={
          Object {
            "maxHeight": "55vh",
            "overflowY": "auto",
          }
        }
      >
        <div
          className="container-fluid"
        >
          <div
            className="table-responsive"
          >
            <table
              className="table"
              style={
                Object {
                  "fontSize": "0.8em",
                }
              }
            >
              <thead>
                <tr>
                  <th>
                    Car
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr
                  className="clickableCell"
                >
                  <td>
                    <span>
                       
                    </span>
                    Audi R8 LMS GT3
                  </td>
                </tr>
                <tr
                  className="success clickableCell"
                >
                  <td>
                    <span
                      className="glyphicon glyphicon-star"
                    />
                    <span>
                       
                    </span>
                    Audi 90 GTO
                  </td>
                </tr>
                <tr
                  className="clickableCell"
                >
                  <td>
                    <span
                      className="glyphicon glyphicon-star"
                    />
                    <span>
                       
                    </span>
                    Audi RS 3 LMS
                  </td>
                </tr>
                <tr
                  className="success clickableCell"
                >
                  <td>
                    <span>
                       
                    </span>
                    BMW M4 GT3
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-primary"
          onClick={[Function]}
          type="button"
        >
          Close
        </button>
      </div>
    </div>
  </div>
</div>
`;
